iocore/eventsystem/SocketManager.cc

```diff
@@ -108,3 +111,36 @@ SocketManager::close(int s)
   } while (res == -EINTR);
   return res;
 }
+
+bool
+SocketManager::fastopen_supported()
```

Comment:
There's no need to enforce the constraint that it can only be called from reconfiguration. We can reasonably make this check from lots of different places. I definitely agree we don't want to check every time we connect a VC.